mbed/mbed.py

```diff
@@ -389,10 +389,13 @@ def fetch_rev(url, rev):
         try:
             if not os.path.exists(rev_file):
                 action("Downloading library build \"%s\" (might take a while)" % rev)
-                outfd = open(rev_file, 'wb')
                 inurl = urlopen(url)
-                outfd.write(inurl.read())
-                outfd.close()
+                with open(rev_file, 'wb') as outfd:
+                    data = None
+                    while data != '':
+                        # Download and write the data in 1 MB chunks
+                        data = inurl.read(1024 * 1024)
+                        outfd.write(data)
         except:
```

Comment:
Perhaps you could add at least:
```suggestion
        except Exception:
```

While you're here?